### PR TITLE
Fix : App crashes sometimes in case of websearch and rss action type

### PR DESCRIPTION
**Pull request by amitiwary999**
Fixes issue #705 

Changes:

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public void onPos(final SourceContent sourceContent, boolean b) {
                         if (imageList == null || imageList.size() == 0) {
                             holder.previewImageView.setVisibility(View.GONE);
                         } else {
-                            Glide.with(context).load(imageList.get(0))
+                            Glide.with(context.getApplicationContext()).load(imageList.get(0))
```
Comment (mayank408) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java]: Why are you using a different context. This is definitely not the cause of the crash.

Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java]: I searched for cause. I found that use getApplicationContext. It solve the problem

**Comment by amitiwary999** [comment]
@chiragw15 review please.

**Comment by chiragw15** [comment]
@amitiwary999 This is a very famous issue in glide library. Although using application context fixes this issue but it is risky and not effective (ref : [link](https://github.com/bumptech/glide/issues/1484#issuecomment-292153414) . Alternate way is https://github.com/bumptech/glide/issues/803#issuecomment-163318397 . Try this. If doesn't work, application context is fine.

**Comment by amitiwary999** [comment]
I tried but currently I found only this solution.

**Comment by chiragw15** [comment]
@amitiwary999 https://github.com/bumptech/glide/issues/803#issuecomment-163318397 Didn't this work? It has many upvotes, should work I guess.

File: app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java (hunk 1 above)
Comment (the-dagger) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java]: @amitiwary999 getting the application context is definitely a bad idea.
Try using the context attached with the holder's widgets. `holder.reviewImageVIew.getContext()`

If this still gives you issues, you might want to switch to Picasso instead.
It works perfectly fine without any such issues.

Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java]: :+1:  ok I will try

**Comment by amitiwary999** [comment]
@chiragw15 @mayank408 @chashmeetsingh review. I changed the code as @the-dagger suggested.

File: app/build.gradle [hunk 2]
```diff
@@ -67,6 +67,9 @@ dependencies {
     compile 'com.github.bumptech.glide:glide:3.7.0'
```
Comment (chiragw15) [inline app/build.gradle]: remove this if you are using picasso

**Comment by chiragw15** [comment]
@amitiwary999 App is crashing continuously with this log. Can you look at it once.
```
FATAL EXCEPTION: main
Process: org.fossasia.susi.ai, PID: 30149
java.lang.IllegalStateException: Center crop requires calling resize with positive width and height.
  at com.squareup.picasso.Request$Builder.build(Request.java:458)
  at com.squareup.picasso.RequestCreator.createRequest(RequestCreator.java:686)
  at com.squareup.picasso.RequestCreator.into(RequestCreator.java:645)
  at com.squareup.picasso.RequestCreator.into(RequestCreator.java:601)
  at org.fossasia.susi.ai.adapters.recycleradapters.ChatFeedRecyclerAdapter$7.onPos(ChatFeedRecyclerAdapter.java:599)
  at com.leocardz.link.preview.library.TextCrawler$GetCode.onPostExecute(TextCrawler.java:64)
  at com.leocardz.link.preview.library.TextCrawler$GetCode.onPostExecute(TextCrawler.java:43)
  at android.os.AsyncTask.finish(AsyncTask.java:651)
  at android.os.AsyncTask.-wrap1(AsyncTask.java)
  at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:668)
  at android.os.Handler.dispatchMessage(Handler.java:102)
  at android.os.Looper.loop(Looper.java:148)
  at android.app.ActivityThread.main(ActivityThread.java:5443)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
```

**Comment by amitiwary999** [comment]
@chiragw15 done.

**Comment by mariobehling** [comment]
@mayank408 @amitiwary999 Please follow up on your earlier exchange and decide if this is good to go or not.

**Comment by amitiwary999** [comment]
@ mariobehling which one? Which issue?

**Pull request merged**